patches/lua-tree.patch

```diff
@@ -148,8 +148,43 @@ index a590f3f..f87343c 100644
      { route = { type = "foreign", reference = "routes", default = null, on_delete = "cascade", }, },
      { service = { type = "foreign", reference = "services", default = null, on_delete = "cascade", }, },
      { consumer = { type = "foreign", reference = "consumers", default = null, on_delete = "cascade", }, },
+diff --git a/lua-tree/share/lua/5.1/kong/db/schema/entities/routes.lua b/lua-tree/share/lua/5.1/kong/db/schema/entities/routes.lua
+index db0696f..43c7f17 100644
+--- a/lua-tree/share/lua/5.1/kong/db/schema/entities/routes.lua
++++ b/lua-tree/share/lua/5.1/kong/db/schema/entities/routes.lua
+@@ -1,6 +1,4 @@
+ local typedefs = require("kong.db.schema.typedefs")
+-local atc = require("kong.router.atc")
+-local router = require("resty.router.router")
+ local deprecation = require("kong.deprecation")
+ 
+ local kong_router_flavor = kong and kong.configuration and kong.configuration.router_flavor
+@@ -42,23 +40,6 @@ if kong_router_flavor == "expressions" then
+       { expression = { type = "string", required = true }, },
+       { priority = { type = "integer", required = true, default = 0 }, },
+     },
+-
+-    entity_checks = {
+-      { custom_entity_check = {
+-        field_sources = { "expression", "id", },
+-        fn = function(entity)
+-          local s = atc.get_schema()
+-          local r = router.new(s)
+-
+-          local res, err = r:add_matcher(0, entity.id, entity.expression)
+-          if not res then
+-            return nil, "Router Expression failed validation: " .. err
+-          end
+-
+-          return true
+-        end,
+-      } },
+-    },
+   }
```

Comment:
Please add this to the patch description like we do for all other changes.

Reply:
Ah, I did log these things, but it must have got reverted while I was making changes. Will add it back.